### DISCUSSION TITLE
USB MIDI

### DISCUSSION THREAD

**Discussion by Lucmunoz** [shanteacontrols/OpenDeck#120] [Q&A] [answered]
Hello Sr.
My name is Lucas and I'm Writing from Chile. I've been looking at your proyect and i'm wondering how did you make the USB MIDI function work using LUFA USB framework on a blackpill STM32F401...

I know that my question is not fully concerned on your project but I'll really apreciate your advice.

Thanks so much!!

**Comment by paradajz** [comment]
I'm not sure what exactly do you want to know. I've just cherry-picked some data types from LUFA to make USB descriptors for STM32 and AVR consistent.

https://github.com/shanteacontrols/OpenDeck/tree/master/src/board/common/comm/usb/descriptors/types

**Reply by Lucmunoz** [reply]
I would like to have USB MIDI on a blackpill. I Haven't found a library that works on the 401 boards They only work on Bluepills.  I've used LUFA on a UNO Board and i was wondering if I could use it on the blackpill as well. That's what i'm loking for... Thanks!

**Comment by paradajz** [comment, marked as answer]
USB stuff for STM32 is here: https://github.com/shanteacontrols/OpenDeck/tree/master/src/board/arch/stm32/comm/usb

**Reply by Lucmunoz** [reply]
So.. Can I use that library? Thanks for your orientation... my knowledge about programming is basic. I mainly use Arduino.

**Reply by Lucmunoz** [reply]
Excuse me Sr... Can you explain me if there is a way to use this as a standalone library To get the USB MIDI función on other projects? Thnak you so much.

**Reply by paradajz** [reply]
Out of the box - no.
